--- "/home/kist/\353\260\224\355\203\225\355\231\224\353\251\264/MJPC/mujoco_mpc/build/_deps/menagerie-src/franka_emika_panda/panda.xml"	2025-11-05 10:50:27.812845302 +0900
+++ "/home/kist/\353\260\224\355\203\225\355\231\224\353\251\264/MJPC/mujoco_mpc/build/_deps/menagerie-src/franka_emika_panda/panda_modified.xml"	2025-11-05 13:12:13.767318446 +0900
@@ -1,7 +1,8 @@
 <mujoco model="panda">
   <compiler angle="radian" meshdir="assets" autolimits="true"/>
 
-  <option integrator="implicitfast"/>
+  <!-- <option integrator="implicitfast"/> -->
+  <option timestep = "0.001"/>
 
   <default>
     <default class="panda">
@@ -215,9 +216,10 @@
                       <geom mesh="hand_3" material="white" class="visual"/>
                       <geom mesh="hand_4" material="off_white" class="visual"/>
                       <geom mesh="hand_c" class="collision"/>
+                      <site name="hand_site" pos="0 0 0"/>
                       <body name="left_finger" pos="0 0 0.0584">
                         <inertial mass="0.015" pos="0 0 0" diaginertia="2.375e-6 2.375e-6 7.5e-7"/>
-                        <joint name="finger_joint1" class="finger"/>
+                        <!-- <joint name="finger_joint1" class="finger"/> -->
                         <geom mesh="finger_0" material="off_white" class="visual"/>
                         <geom mesh="finger_1" material="black" class="visual"/>
                         <geom mesh="finger_0" class="collision"/>
@@ -229,7 +231,7 @@
                       </body>
                       <body name="right_finger" pos="0 0 0.0584" quat="0 0 0 1">
                         <inertial mass="0.015" pos="0 0 0" diaginertia="2.375e-6 2.375e-6 7.5e-7"/>
-                        <joint name="finger_joint2" class="finger"/>
+                        <!-- <joint name="finger_joint2" class="finger"/> -->
                         <geom mesh="finger_0" material="off_white" class="visual"/>
                         <geom mesh="finger_1" material="black" class="visual"/>
                         <geom mesh="finger_0" class="collision"/>
@@ -250,19 +252,19 @@
     </body>
   </worldbody>
 
-  <tendon>
+  <!-- <tendon>
     <fixed name="split">
       <joint joint="finger_joint1" coef="0.5"/>
       <joint joint="finger_joint2" coef="0.5"/>
     </fixed>
-  </tendon>
+  </tendon> -->
 
-  <equality>
+  <!-- <equality>
     <joint joint1="finger_joint1" joint2="finger_joint2" solimp="0.95 0.99 0.001" solref="0.005 1"/>
-  </equality>
+  </equality> -->
 
   <actuator>
-    <general class="panda" name="actuator1" joint="joint1" gainprm="4500" biasprm="0 -4500 -450"/>
+    <!-- <general class="panda" name="actuator1" joint="joint1" gainprm="4500" biasprm="0 -4500 -450"/>
     <general class="panda" name="actuator2" joint="joint2" gainprm="4500" biasprm="0 -4500 -450"
       ctrlrange="-1.7628 1.7628"/>
     <general class="panda" name="actuator3" joint="joint3" gainprm="3500" biasprm="0 -3500 -350"/>
@@ -271,13 +273,20 @@
     <general class="panda" name="actuator5" joint="joint5" gainprm="2000" biasprm="0 -2000 -200" forcerange="-12 12"/>
     <general class="panda" name="actuator6" joint="joint6" gainprm="2000" biasprm="0 -2000 -200" forcerange="-12 12"
       ctrlrange="-0.0175 3.7525"/>
-    <general class="panda" name="actuator7" joint="joint7" gainprm="2000" biasprm="0 -2000 -200" forcerange="-12 12"/>
+    <general class="panda" name="actuator7" joint="joint7" gainprm="2000" biasprm="0 -2000 -200" forcerange="-12 12"/> -->
+    <motor class="panda" name="actuator1" joint="joint1" gear="1" ctrlrange="-87 87" forcerange="-87 87"/>
+    <motor class="panda" name="actuator2" joint="joint2" gear="1" ctrlrange="-87 87" forcerange="-87 87"/>
+    <motor class="panda" name="actuator3" joint="joint3" gear="1" ctrlrange="-87 87" forcerange="-87 87"/>
+    <motor class="panda" name="actuator4" joint="joint4" gear="1" ctrlrange="-87 87" forcerange="-87 87"/>
+    <motor class="panda" name="actuator5" joint="joint5" gear="1" ctrlrange="-12 12" forcerange="-12 12"/>
+    <motor class="panda" name="actuator6" joint="joint6" gear="1" ctrlrange="-12 12" forcerange="-12 12"/>
+    <motor class="panda" name="actuator7" joint="joint7" gear="1" ctrlrange="-12 12" forcerange="-12 12"/>
     <!-- Remap original ctrlrange (0, 0.04) to (0, 255): 0.04 * 100 / 255 = 0.01568627451 -->
-    <general class="panda" name="actuator8" tendon="split" forcerange="-100 100" ctrlrange="0 255"
-      gainprm="0.01568627451 0 0" biasprm="0 -100 -10"/>
+    <!-- <general class="panda" name="actuator8" tendon="split" forcerange="-100 100" ctrlrange="0 255"
+      gainprm="0.01568627451 0 0" biasprm="0 -100 -10"/> -->
   </actuator>
 
-  <keyframe>
+  <!-- <keyframe>
     <key name="home" qpos="0 0 0 -1.57079 0 1.57079 -0.7853 0.04 0.04" ctrl="0 0 0 -1.57079 0 1.57079 -0.7853 255"/>
-  </keyframe>
+  </keyframe> -->
 </mujoco>
